test/extended/cpu_partitioning/crio.go

```diff
@@ -0,0 +1,314 @@
+
+func getExpectedCPUSetConstraints(ctx context.Context, oc *exutil.CLI, isClusterPartitioned bool) (map[string]string, error) {
+	// Default behavior is that CPUSets are empty, which means wide open to the host.
+	// There should be no CPUSet constraint if no PerformanceProfile is present.
+	nodeRoleCPUSet := map[string]string{nodeMasterLabel: "", nodeWorkerLabel: ""}
+
+	if !isClusterPartitioned {
+		return nodeRoleCPUSet, nil
+	}
+
+	// Performance profiles dictate constrained CPUSets, we gather them here to compare.
+	performanceProfiles, err := oc.AdminDynamicClient().
+		Resource(schema.GroupVersionResource{
+			Resource: "performanceprofiles",
+			Group:    "performance.openshift.io",
+			Version:  "v2"}).Namespace("").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error listing performance profiles: %w", err)
+	}
+
+	for _, profile := range performanceProfiles.Items {
+
+		selector, found, err := unstructured.NestedStringMap(profile.Object, "spec", "nodeSelector")
+		if err != nil {
+			return nil, fmt.Errorf("error getting nodeSelector from PerformanceProfile: %w", err)
+		}
+		if !found {
+			return nil, fmt.Errorf("expected spec.nodeSelector to be found in PerformanceProfile(%s)", profile.GetName())
+		}
+
+		reservedCPUSet, found, err := unstructured.NestedString(profile.Object, "spec", "cpu", "reserved")
+		if err != nil {
+			return nil, fmt.Errorf("error getting reservedCPUSet from PerformanceProfile: %w", err)
+		}
+		if !found {
+			return nil, fmt.Errorf("expected spec.reserved to be found in PerformanceProfile(%s)", profile.GetName())
+		}
+
+		// Single Node may include both worker and master labels for node selection,
+		// we match on Master first to account for that, since Master get's preference when both
+		// labels are present.
+		if _, ok := selector[nodeMasterLabel]; ok {
+			nodeRoleCPUSet[nodeMasterLabel] = reservedCPUSet
+		} else if _, ok := selector[nodeWorkerLabel]; ok {
+			nodeRoleCPUSet[nodeWorkerLabel] = reservedCPUSet
+		}
+	}
+
+	return nodeRoleCPUSet, nil
+}
+
+// Execute collection script on machine-config-daemon of every node. Marshal the results to get a list of containers
+// running in the cluster.
+func collectContainerInfo(cfg *rest.Config, c kubernetes.Interface, node corev1.Node) ([]crioContainerData, error) {
+
+	listOptions := metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}).String(),
+		LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}).String(),
+	}
+
+	pods, err := c.CoreV1().Pods(namespaceMachineConfigOperator).List(context.TODO(), listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(pods.Items) < 1 {
+		return nil, fmt.Errorf("failed to get machine-config-daemon pod for the node %q", node.Name)
+	}
+
+	pod := &pods.Items[0]
+
+	var out []byte
+	getInfo := []string{
+		"chroot",
+		"/rootfs",
+		"/bin/bash",
+		"-c", collectionScript}
+
+	ctx := context.TODO()
```

Comment:
It's usually better to pass the context from the caller, although here it's just a test so no biggie.

Reply:
Sure thing, updated.